Source/DafnyLanguageServer/Workspace/NotificationPublisher.cs

```diff
@@ -87,18 +87,18 @@ public class NotificationPublisher : INotificationPublisher {
     }
 
     private CompilationStatus GetGlobalProgress(IdeState state) {
-      var hasResolutionDiagnostics = state.ResolutionDiagnostics.Values.SelectMany(x => x).
-        Any(d => d.Severity == DiagnosticSeverity.Error);
+      var errors = state.ResolutionDiagnostics.Values.SelectMany(x => x).
+        Where(d => d.Severity == DiagnosticSeverity.Error).ToList();
       if (state.Compilation is CompilationAfterResolution) {
-        if (hasResolutionDiagnostics) {
+        if (errors.Any(d => d.Source == MessageSource.Resolver.ToString())) {
```

Comment:
Minor thing: Don't we have a way to cache all these .ToString() ? I hope they are compiled as constant strings because otherwise it feels it should be refactored in a constant somewhere.

Reply:
> I hope they are compiled as constant strings

Seems likely to me